python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1107,11 +1108,46 @@ def wipe_assets(self, asset_keys):
 
     # event subscriptions
 
-    def get_event_log_handler(self):
+    def _get_yaml_python_handlers(self):
+        handlers = []
+
+        if self._settings:
+            handlers_dict = self._settings.get("python_logs", {}).get("handlers", {})
+
+            if handlers_dict:
+                # not sure what configuration settings we'll have in python_logs.
+                # we only want to feed in handlers to logging.config.DictConfigurator
+                payload = {
+                    "handlers": dict(handlers_dict)
+                }  # DictConfigurator mutates objects, so we make a copy here
+                dict_configurator = logging.config.DictConfigurator(payload)
+
+                # dict_configurator.config.get from the python logging module converts
+                # strings to their expected types
+                formatted_handlers_attr = dict_configurator.config.get("handlers", {})
+                for name in sorted(formatted_handlers_attr):
+                    handler = dict_configurator.configure_handler(formatted_handlers_attr[name])
+
+                    # initialize dummy format so we can see logs
+                    formatter = logging.Formatter(
```

Comment:
Does this mean that if you don't specify a formatter, then you will get no output (assuming this line doesn't exist)? I hadn't really thought of this issue, but now that you've brought it up, this seems like something people might want to configure themselves (for example, a lot of use cases might want a json-formatted output).

This implies that we should probably let people configure formatters in the same place that they configure handlers (just as they can in a normal dictionary config). If that's the case, then we might want to rethink this implementation a bit, as it might be a bit too unmaintainable to copy the DictConfigurator implementation for both the handler stuff AND the formatter stuff.

I'm imagining something where a dagster.yaml thing looks something like: 

```
python_logs:
    python_logging_level: INFO
    managed_logs:
        - "root"
    dagster_handler_config:
        formatters:
             ...
        handlers:
             ...
```

From there, if we just do logging.dictConfig(dagster_handler_config), we know the logging._handlers dictionary will be populated with instantiated handlers, each named by a key in the dagster_handler_config["handlers"] dictionary, so we can accurately get a list of those back, without having to reimplement any of this logic.

However, I'm not completely sold on this solution (seems a bit gross to be accessing private module members), so we should probably talk through this a bit more.

Reply:
I just double checked about this, I was actually mistaken about the handler not outputting. If a formatter is not added, then the default behavior is to print only the event record message (which looks something like this):
```
log_pipeline - 42ceda34-3736-4318-8eb0-1b583d5e603e - log_solid - Hello world
```
However, I agree that users will want to configure their own formatters. I'll keep thinking about another solution, and we can sync on this.